docs/proposals/hpa/federated-hpa.md

```diff
@@ -0,0 +1,286 @@
+
+#### Story 2
+For an application developer, I create a HPA CR for the application running on Karmada with FederatedHPA enabled.
+```
+target cpu util 30%
+min replica 3
+max replica 100
+```
+Suddenly, one of the member clusters which my application running on stops working and can't scale up new pods. Unfortunately, a request burst is comming into the application. The CPU util of pods becomes higher than 30%. It will need 100 Pods totally to take the request burst. I hope the Karmada FederatedHPA can scale up new pods in other healthy clusters.
+
+#### Story 3
+As an administrator of the Karmada&Kubernetes platform, I receive an alert that the Karmada control plane stops working and any requests to the Karmada control plane are failed. There are many applications running on the platform heavilly depend on the HPA to handle the unpredictable burst of requests. The chance of RCA occurred becomes really high if the system can't tolerate the failure of federation control plane. So I hope the Karmada FederatedHPA can scale in the member clusters even if the Karmada control plane is down.
+
+### Notes/Constraints/Caveats (Optional)
+1. The workloads/pods in different member clusters selected by the same HPA CR/resource share the load of the application equally. For example, 10 pods of the application are spread into two member clusters with distribution `cluster1: 3 pods, cluster2: 7 pods`, so the 3 pods in cluster1 take 3/10 of total requests and 7 pods in cluster2 take 7/10 of total requests. Scenarios don't meet the restriction are not considered in this proposal.
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? 
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+### Architecture
+![Karmada FederatedHPA Architecture](statics/karmadafederatedhpa-arch.drawio.png)
+
+There are no new CRDs or resources introduced in this design. All the core functions are implemented in the `FederatedHPAController`.
+1. The Kubernetes HPA components are still used in the member cluster and can work standalone.
+1. The FederatedHPAController is responsible for the purposes
+   1. Watch the HPA resource and `PropagationPolicy/ResourceBinding` corresponding to the `Workload`, to learn which clusters the HPA resource should propagated to and what weight the workloads should be spread between clusters.
```

Comment:
```suggestion
   1. Watch the HPA resource and `ResourceBinding` corresponding to the `Workload`, to learn which clusters the HPA resource should propagated to and what weight the workloads should be spread between clusters.
```
I think the `FederatedHPAController` don't need to watch `PropagationPolicy`, it can get the workload's metadata and target cluster from the `ResourceBinding`.